sdl_android/src/main/java/com/smartdevicelink/protocol/SdlProtocol.java

```diff
@@ -0,0 +1,1457 @@
+            return true;
+        }else{
+            //Since the only service type that should reach this point are ones that require a high
+            //bandwidth, true can be returned if a secondary transport is a high bandwidth transport
+            return isSecondaryTransportAvailable(true);
+        }
+    }
+
+    /**
+     * Checks to see if a secondary transport is available for this session
+     * @param onlyHighBandwidth if only high bandwidth transports should be included in this check
+     * @return true if any connected or potential transport meets the criteria to be a secondary
+     *         transport
+     */
+    private boolean isSecondaryTransportAvailable(boolean onlyHighBandwidth){
+        if (supportedSecondaryTransports != null) {
+            for (TransportType supportedSecondary : supportedSecondaryTransports) {
+                if(!onlyHighBandwidth || supportedSecondary == TransportType.USB || supportedSecondary == TransportType.TCP) {
+                    if (transportManager.isConnected(supportedSecondary, null)) {
+                        //A supported secondary transport is already connected
+                        return true;
+                    } else if (secondaryTransportParams != null && secondaryTransportParams.containsKey(supportedSecondary)) {
+                        //A secondary transport is available to connect to
+                        return true;
+                    }
+                }
+            }
+        }
+        // No supported secondary transports
+        return false;
+    }
+
+
+    /**
+     * If there was a TransportListener attached to the supplied multiplex config, this method will
+     * call the onTransportEvent method.
+     */
+    private void notifyDevTransportListener (){
+        if(transportConfig.getTransportListener() != null && transportManager != null) {
+            transportConfig.getTransportListener().onTransportEvent(transportManager.getConnectedTransports(), isTransportForServiceAvailable(SessionType.PCM),isTransportForServiceAvailable(SessionType.NAV));
+        }
+    }
+
+    /**
+     * Retrieves the preferred transport for the given connected transport
+     * @param preferredList the list of preferred transports (primary or secondary)
+     * @param connectedTransports the current list of connected transports
+     * @return the preferred connected transport
+     */
+    private TransportRecord getPreferredTransport(List<TransportType> preferredList, List<TransportRecord> connectedTransports) {
+        for (TransportType transportType : preferredList) {
+            for(TransportRecord record: connectedTransports) {
+                if (record.getType().equals(transportType)) {
+                    return record;
+                }
+            }
+        }
+        return null;
+    }
+
+    public TransportRecord getPreferredPrimaryTransport(List<TransportRecord> transports){
```

Comment:
This can be private

Reply:
Only called once, removing method and using more generic method instead